### ISSUE TITLE
[a11y]: Overflow Menu - JAWS is not announcing/skips the disabled options present in Menu 

### ISSUE THREAD

**Issue by 2nikhiltom** [carbon-design-system/carbon#16533]
### Package

@carbon/react

### Package version

v1.56.0

### React version

18.02

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS

### Description

In the [storybook](https://react.carbondesignsystem.com/?path=/story/components-overflowmenu--default) example 
While navigating through the options in the dropdown list, the user is aware that there are six list elements. JAWS announces each option individually during navigation, skipping the disabled option, which is "option 4 of 6". The JAWS announcements are as follows:

- "Overflow-menu menu"
- "Stop app"
- "1 of 6. To move through items, press up or down arrow."
- "Restart app"
- "2 of 6"
- "Rename app"
- "3 of 6"
- "Edit routes and access"
- "5 of 6"
- "Delete app"
- "6 of 6"
- "Edit routes and access"
- "5 of 6"

Open 4 is skipped by SR (Not announced )
<img width="459" alt="image" src="https://github.com/carbon-design-system/carbon/assets/63502271/24ef24b5-cfd7-452b-b628-f0ef01dcbe85">

When the state of an element i.e. **disabled**, selected, expanded, currently active etc. is not conveyed for the screen reader user, it can be misleading for the user. . this impacts the user interaction.

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-overflowmenu--default

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by VishwaKumar** [comment]
This defect is still present in latest version of carbon react.

**Comment by tombrunet** [comment]
@VishwaKumar This is expected behavior. Try with a native select, like:
```
<select name="options" id="options">
  <option value="" disabled selected>-- Select an option --</option>
  <option value="option1">Option 1</option>
  <option value="option2" disabled>Option 2 (Unavailable)</option>
  <option value="option3">Option 3</option>
  <option value="option4" disabled>Option 4 (Unavailable)</option>
  <option value="option5">Option 5</option>
</select>
```

If you look at the accessibility tree of the carbon storybook, you'll see that the information is exposed to the screen reader. The screen reader is choosing not to expose that to the user.

<img width="216" height="197" alt="Image" src="https://github.com/user-attachments/assets/808a9d18-8185-46dc-b555-a02fc80dfb35" />

**Comment by heloiselui** [comment]
I’m closing this issue in favor of what was mentioned above, thanks @tombrunet.
